Automattic/jetpack
# VideoPress: Add rating and privacy_settings meta keys for VideoPress attachments

Relates to #26168.

#### Changes proposed in this Pull Request:

* Start to save a copy of the `rating` and the `privacy_setting` of the videos on their own meta keys (`videopress_rating` and `videopress_privacy_setting`); by doing this, we can easily filter the videos using the fields, what we could not do using the serialized data from the `_wp_attachment_metadata`
* Add code to make sure this new meta data will be filled for the videos uploaded before this change, so they get migrated to the new format; the update is triggered when the details for one or more videos are requested and only happens if the value for the new meta keys is not set
 
#### Other information:


#### Jetpack product discussion



N/A.

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* To **test the new meta fields**, we need to update the value for the privacy and the rating and then check it on the database to make sure the new meta keys are set
* To update one of these fields, do a POST request to the `/meta` endpoint:

```
curl -X POST -u "user:pass" http://your-site.com/wp-json/wpcom/v2/videopress/meta -H 'Content-Type: application/json' -d '{"id": {ID},"rating": "PG-13","privacy_setting":"0" }' | jq
```
* You can also update the privacy using the VideoPress client (but not the rating, for now):

![image](https://user-images.githubusercontent.com/6760046/195195558-8eca673e-7411-44bf-be32-dfde05b00982.png)

* To check the database meta data for this specific video, query:

```
SELECT * FROM `wp_postmeta` where post_id = {ID};
```

* You should see the two meta keys, `videopress_rating` and `videopress_privacy_setting`:

![image](https://user-images.githubusercontent.com/6760046/195194476-55245841-dade-4846-867a-c0945568356f.png)

* To **test the automatic migration of the values** for existing videos, just open your VideoPress Library (`Jetpack > VideoPress`) and then verify the database meta keys for all the videos showed on the library; everytime one or more videos are loaded, the meta keys are checked and then updated in the case they are missing
* You can also test for a specific video by manually requesting the details for it:

```
curl -X GET -u "user:pass" http://your-site.com/wp-json/wp/v2/media/{ID}
```

* After this request, the new meta fields should exist on the database for the given video
